lib/sypex_geo/result.rb

```diff
@@ -20,15 +20,15 @@ def initialize(position, database)
     end
 
     def city
-      @city ||= @database.read_city(@position)
+     @database.read_city(@position).nil? ? nil : @city ||= @database.read_city(@position)
```

Comment:
This will break memoisation. Please, consider to use `defined?` check here, like:
```ruby
def city
  return @city if defined?(@city)
  @city = @database.read_city(@position)
end
```